cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -401,6 +401,9 @@ Result<S3Options> S3Options::FromUri(const Uri& uri, std::string* out_path) {
       options.scheme = kv.second;
     } else if (kv.first == "endpoint_override") {
       options.endpoint_override = kv.second;
+    } else if (kv.first == "allow_delayed_open") {
```

Comment:
FYI Other existing options are missing, for example: `background_writes`, `check_directory_existence_before_creation`, `request_timeout`...

Reply:
Yes, please do if you are interested (as a separate PR though).

Reply:
Thanks !